src/appleseed/renderer/modeling/entity/entityfactoryregistrar.cpp

```diff
@@ -72,6 +72,50 @@ EntityFactoryRegistrar::~EntityFactoryRegistrar()
     delete impl;
 }
 
+void EntityFactoryRegistrar::discover_plugins(
+    const foundation::SearchPaths&          search_paths){
+    namespace bf = boost::filesystem;
+
+    // Iterate over all search paths.
+    for (size_t i = 0, e = search_paths.get_path_count(); i < e; ++i)
+    {
+        bf::path search_path(search_paths.get_path(i));
+
+        // Make the search path absolute if it isn't already.
+        if (!search_path.is_absolute() && search_paths.has_root_path())
+            search_path = search_paths.get_root_path().c_str() / search_path;
+
+        // Only consider directories.
+        if (!bf::exists(search_path) || !bf::is_directory(search_path))
+            continue;
+
+        // Iterate over all files in this directory.
+        for (bf::directory_iterator j(search_path), f; j != f; ++j)
+        {
+
+            // Only consider shared library files.
+            if (!bf::is_regular_file(*j) ||
+                j->path().extension() != foundation::SharedLibrary::get_default_file_extension())
+                continue;
+
+            const std::string plugin_path = j->path().string();
+            
+            //load plugin into memory 
```

Comment:
Follow the commenting style:
- Insert a space after `//`
- Capitalize the first letter of the sentence.
- End the sentence with a point.